lib/Autoloader.php

```diff
@@ -58,6 +63,51 @@ public static function removePackage(Package $package)
 	}
 
 	// }}}
+	// {{{ public static function addFinalPackage()
+
+	/**
+	 * Sets an autoloader package as the last package to check against
+	 *
+	 * @param Package $package the autoloader package to add.
+	 *
+	 * @return void
+	 */
+	public static function setFinalPackage(Package $package)
+	{
+		self::$final_package = $package;
+	}
+
+	// }}}
+	// {{{ public static function removeFinalPackage()
+
+	/**
+	 * Removes the final autoloader package from the autoloader
+	 *
+	 * @return void
+	 */
+	public static function removeFinalPackage()
+	{
+		self::$final_package = null;
+	}
+
+	// }}}
+	// {{{ public static function getPackages()
+
+	/**
+	 * @return array
+	 */
+	public static function getPackages()
+	{
+		$packages = self::$packages;
```

Comment:
This will be called once per class on every request. This implementation works but we should make it more efficient. Calling `setFinalPackage()` should update the `self::$packages` array directly and `getPackages()` should be removed.